packages/react/src/components/FileSelect/FileSelect.tsx

```diff
@@ -129,6 +129,11 @@ export const useFileSelect = (
           onSelect?.([...event.target.files]);
         }
 
+        // Reset the input value to allow re-selecting the same file
```

Comment:
another approach is to add a `key` prop to `FileSelect` component and dynamically update it when onChange is called to make sure it re-renders